components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2044,17 +2044,28 @@ private Set<org.wso2.carbon.user.core.common.User> getMultiAttributeFilteredUser
     private int getMaxLimit(String domainName) {
 
         int givenMax = UserCoreConstants.MAX_USER_ROLE_LIST;
-        if (StringUtils.isEmpty(domainName)) {
-            domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
-        }
 
-        if (carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
-                .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST) != null) {
-            givenMax = Integer.parseInt(carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
-                    .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST));
+        try {
+            if (StringUtils.isEmpty(domainName)) {
+                domainName = carbonUM.getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig.
+                        PROPERTY_DOMAIN_NAME);
+                if (log.isDebugEnabled()) {
+                    log.debug("Primary user store DomainName picked as " + domainName);
+                }
+            }
+            if (carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
+                    .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST) != null) {
+                givenMax = Integer.parseInt(carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
+                        .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST));
+            }
+
+            return givenMax;
+
+        } catch(NullPointerException e) {
```

Comment:
NullPointerException is a run-time exception which is not recommended to catch it, but instead avoid it with a null check

Reply:
Fixed with : [b6138a4ccb87d](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/451/commits/b6138a4ccb87d5a03cbe5bb52a1cb61c1e3409db)